src/system.h

```diff
@@ -278,55 +283,76 @@ struct System {
-                            uchar expected[] = {0x89, 'P', 'N', 'G', '\r', '\n', 0x1A, '\n'};
-                            if (memcmp(header, expected, 8)) return _(L"Corrupt PNG header.");
-                            dis.BigEndianOrdered(true);
-                            for (;;)  // Skip all chunks.
-                            {
-                                wxInt32 len = dis.Read32();
-                                char fourcc[4];
-                                fis.Read(fourcc, 4);
-                                fis.SeekI(len, wxFromCurrent);  // skip data
-                                dis.Read32();                   // skip CRC
-                                if (memcmp(fourcc, "IEND", 4) == 0) break;
+                            if (beforeimage == wxInvalidOffset) return _(L"Cannot tell/seek document?");
+                            fis.SeekI(beforeimage);
+                            switch(imt) {
+                                case wxBITMAP_TYPE_PNG: {
+                                    uchar header[8];
+                                    fis.Read(header, 8);
+                                    uchar expected[] = {0x89, 'P', 'N', 'G', '\r', '\n', 0x1A, '\n'};
+                                    if (memcmp(header, expected, 8)) return _(L"Corrupt PNG header.");
+                                    dis.BigEndianOrdered(true);
+                                    for (;;)  // Skip all chunks.
+                                    {
+                                        wxInt32 len = dis.Read32();
+                                        char fourcc[4];
+                                        fis.Read(fourcc, 4);
+                                        fis.SeekI(len, wxFromCurrent);  // skip data
+                                        dis.Read32();                   // skip CRC
+                                        if (memcmp(fourcc, "IEND", 4) == 0) break;
+                                    }
+                                    break;
+                                }
+                                case wxBITMAP_TYPE_JPEG: {
+                                    dis.BigEndianOrdered(true);
+                                    for (;;)  // Skip all chunks.
+                                    {
+                                        wxInt32 len = dis.Read32();
+                                        char twocc[2];
+                                        fis.Read(twocc, 2);
+                                        fis.SeekI(len, wxFromCurrent);  // skip data
+                                        dis.Read32();                   // skip CRC
+                                        if (memcmp(twocc, "\xff\xd9", 2) == 0) break;
+                                    }
+                                    break;
+                                }
+                                default: {
+                                    return _(L"Unknown identifer for data");

```

Comment:
make this error a lot more specific, if the user gets this error they have no idea what went wrong